botocore/args.py

```diff
@@ -620,12 +621,15 @@ def _ensure_boolean(self, val):
         else:
             return val.lower() == 'true'
 
-    def _construct_builtin_resolver(self, credentials, client_config):
+    def _construct_builtin_resolver(
+        self, credentials, client_config, uses_builtin_data
+    ):
         credential_builtin_resolver = CredentialBuiltinResolver(
-            credentials, client_config.account_id_endpoint_mode
+            credentials,
+            client_config.account_id_endpoint_mode,
         )
         resolver_map = {'credentials': credential_builtin_resolver}
-        return EndpointBuiltinResolver(resolver_map)
+        return EndpointBuiltinResolver(resolver_map, uses_builtin_data)
```

Comment:
If this is only for the warning, we shouldn't be changing the function signature for this. Let's use a simpler patch, if we resolve a custom credential scope. We'll warn the first time that it may not be honored for that endpoint if one is defined for the supplied credentials.

We should validate this _only_ warns when there's a custom `endpoints.json` in `~/.aws/models`.